### PR TITLE
Master fix ajax abort chm

### PR DESCRIPTION
**Pull request by Gorash**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/core/session.js [hunk 1]
```diff
@@ -344,9 +344,9 @@ var Session = core.Class.extend(mixins.EventDispatcherMixin, {
             }
         };
 
-        return deferred.then(function () {
+        var newDef = deferred.then(function () {
```
Comment (alexkuhn) [inline addons/web/static/src/js/core/session.js]: `var promise = ...`

File: addons/web/static/tests/core/widget_tests.js [hunk 2]
```diff
@@ -377,16 +379,49 @@ QUnit.module('core', {}, function () {
...
-        var widget = new Widget();
+        widget._rpc({route: '/a/route'}).then(function () {
+            assert.ok(true, "The ajax call should be resolve");
+        });
+        def.resolve();
+        def = null;
+
+        widget._rpc({route: '/a/route'}).always(function () {
+            throw Error("Calling _rpc on a destroyed widget should return a " +
+                "deferred that is never resolved nor rejected");
```
Comment (aab-odoo) [inline addons/web/static/tests/core/widget_tests.js]: I don't understand the message: widget is not destroyed yet

File: addons/web/static/tests/helpers/mock_server.js [hunk 3]
```diff
@@ -1031,76 +1040,99 @@ var MockServer = Class.extend({
...
+     *
+     * @private
+     * @param {string} route
+     * @param {Object} args
+     * @returns {Deferred<any>}
+     *          Resolved with the result of the RPC. If the RPC should fail, the
+     *          deferred should either be rejected or the call should throw an
+     *          exception (@see performRpc for error handling).
+     */
+    _performRpc: function (route, args) {
```
Comment (aab-odoo) [inline addons/web/static/tests/helpers/mock_server.js]: _performRPC is overriden in some addons

Comment (Gorash) [reply addons/web/static/tests/helpers/mock_server.js]: no problem for that, it's a simplification for not having deferred in deferred for nothing (uses cpu for nothing)

File: addons/web/static/tests/core/widget_tests.js (hunk 2 above)
Comment (Gorash) [reply addons/web/static/tests/core/widget_tests.js]: We destroy the widget just after, here the exception allows to see that the request is not done if the widget is destroyed.

**Comment by Gorash** [comment]
https://github.com/odoo/odoo/pull/26599

**Pull request closed without merging**